src/main/java/com/redhat/devtools/intellij/tektoncd/actions/DeleteAction.java

```diff
@@ -137,7 +118,71 @@ public void actionPerformed(AnActionEvent anActionEvent, TreePath[] path, Object
                 }
             }
         });
+    }
 
+    private String getTitle(ParentableNode[] elements) {
+        if (elements.length == 1) {
+            return "Delete " + elements[0].getName();
+        } else {
+            return "Delete multiple items";
+        }
+    }
+
+    private String getDeleteText(Tkn tkn, ParentableNode[] elements) {
+        StringBuilder sb = new StringBuilder("Are you sure you want to delete ");
+        if (elements.length == 1) {
+            String name = elements[0].getName();
+            String kind = elements[0].getClass().getSimpleName().toLowerCase().replace("node", "");
+            int usages = getUsages(tkn, kind, name);
+            if (usages > 0) {
+                sb.insert(0, "This " + kind + " is being used by other resources. ");
+            }
+            sb.append(kind + " " + name + " ?");
+        } else {
+            sb.append("the following items?\n");
+            for (ParentableNode element: elements) {
+                String kind = elements[0].getClass().getSimpleName().toLowerCase().replace("node", "");
+                sb.append(element.getName() + getUsagesAsText(tkn, kind, element.getName()) + "\n");
+            }
+        }
+        return sb.toString();
+    }
+
+    private String getDeleteChkText(ParentableNode[] elements) {
+        if (elements.length == 1) {
+            String kind = elements[0].getClass().getSimpleName().toLowerCase().replace("node", "");
+            return mayHaveRelatedResource(elements[0]) ? "Also delete its related resources (" + kind + "Runs)" : "";
+        } else {
+            return "Also delete their related resources (PipelineRuns, TaskRuns..)";
+        }
+    }
+
+    private boolean mayHaveRelatedResource(ParentableNode element) {
+        return element instanceof PipelineNode
+                || element instanceof TaskNode
+                || element instanceof ClusterTaskNode;
+    }
+
+    private String getUsagesAsText(Tkn tkn, String kind, String name) {
+        int usages = getUsages(tkn, kind, name);
+        if (usages == 0) {
+            return "";
+        }
+        return " (found " + usages + " " + (usages == 1 ? "usage" : "usages") + ")";
+    }
+
+    private int getUsages(Tkn tkn, String kind, String name) {
+        if (!(kind.equalsIgnoreCase(KIND_TASK) || kind.equalsIgnoreCase(KIND_CLUSTERTASK))) {
+            return 0;
+        }
+
+        try {
+            List<RefUsage> usages = tkn.findTaskUsages(kind, name);
+            return usages.size();
+        } catch (IOException e) {
+            logger.warn(e.getLocalizedMessage());
```

Comment:
Exception is missing

Reply:
you mean `logger.warn(e.getLocalizedMessage(), e)`?

Reply:
Yes